python/tank_vendor/shotgun_base/utils.py

```diff
@@ -23,27 +24,49 @@ def with_cleared_umask(func):
     """
     Decorator which clears the umask for a method.
 
-    The umask controls default permissions for files
-    and folders. By setting the umask to zero, we can control
-    permissions.
+    The umask is a permissions mask that gets applied
+    whenever new files or folders are created. For I/O methods
+    that have a permissions parameter, it is important that the
+    umask is cleared prior to execution, otherwise the default
+    umask may alter the resulting permissions, for example::
 
-    Example use:
-
-    @with_cleared_umask
-    def my_io_method(param1, param2):
+        def create_folders(path, permissions=0777):
+            log.debug("Creating folder %s..." % path)
+            os.makedirs(path, permissions)
 
-        # do i/o operations in here
+    The 0777 permissions indicate that we want folders to be
+    completely open for all users (a+rwx). However, the umask
+    overrides this, so if the umask for example is set to 0777,
+    meaning that I/O operations are not allowed to create files
+    that are readable, executable or writable for users, groups
+    or others, the resulting permissions on folders created
+    by create folders will be 0, despite passing in 0777 permissions.
+
+    By adding this decorator to the method, we temporarily reset
+    the umask to 0, thereby giving full control to
+    any permissions operation to take place without any restriction
+    by the umask::
+
+        @with_cleared_umask
+        def create_folders(path, permissions=0777):
+            # Creates folders with the given permissions,
+            # regardless of umask setting.
+            log.debug("Creating folder %s..." % path)
+            os.makedirs(path, permissions)
     """
-    def inner(*args, **kwargs):
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
         # set umask to zero, store old umask
         old_umask = os.umask(0)
+        log.debug("Umask cleared")
```

Comment:
I am wondering if the debug message isn't overkill. Does it appear a million times in the logs?